src/fixed-footer-layout.tsx

```diff
@@ -43,43 +43,48 @@ type Props = {
     onChangeFooterHeight?: (heightInPx: number) => void;
 };
 
-const FixedFooterLayout: React.FC<Props> = ({
+const MIN_AVAILABLE_HEIGHT_FOR_FIXED = 200;
+
+const FixedFooterLayout = ({
     isFooterVisible = true,
     footer,
     footerHeight = 'auto',
     footerBgColor = vars.colors.background,
-    containerBgColor,
+    containerBgColor = vars.colors.background,
     children,
     onChangeFooterHeight,
-}) => {
+}: Props): JSX.Element => {
     const [displayElevation, setDisplayElevation] = React.useState(false);
     const containerRef = React.useRef<HTMLDivElement>(null);
+    const {height: contentHeight} = useBoundingRect(containerRef) || {height: 0};
     const {isTabletOrSmaller} = useScreenSize();
     const {platformOverrides} = useTheme();
     const {height: domFooterHeight, ref} = useElementDimensions();
-    const isWithinIFrame = useIsWithinIFrame();
-    const windowHeight = useWindowHeight();
-    const screenHeight = useScreenHeight();
-    const hasContentEnoughVSpace =
-        windowHeight - domFooterHeight > (isWithinIFrame ? windowHeight : screenHeight) / FOOTER_CANVAS_RATIO;
+    const {visualHeight} = useWindowSize();
+    const topDistance = useFixedToTopHeight();
+    const availableHeight = visualHeight - topDistance - domFooterHeight;
+    const footerIsFixed = availableHeight > MIN_AVAILABLE_HEIGHT_FOR_FIXED;
 
     useIsomorphicLayoutEffect(() => {
         onChangeFooterHeight?.(domFooterHeight);
     }, [onChangeFooterHeight, domFooterHeight]);
 
+    const footerHeightStyle = `calc(${safeAreaInsetBottom} + ${domFooterHeight}px)`;
+
     React.useEffect(() => {
         /**
          * There is no elevation in desktop devices and we don't display it in acceptance tests or when the
          * content's height is too small, so we avoid unnecesary calculations in these cases.
          */
-        if (!isTabletOrSmaller || isRunningAcceptanceTest(platformOverrides) || hasContentEnoughVSpace) {
+        if (!isTabletOrSmaller || isRunningAcceptanceTest(platformOverrides) || !footerIsFixed) {
+            setDisplayElevation(false);
             return;
         }
 
         const scrollable = getScrollableParentElement(containerRef.current);
 
         const shouldDisplayElevation = () =>
-            hasScroll(scrollable) && getScrollDistanceToBottom(scrollable) > 1; // This is 1 and not 0 because a weird bug with Safari
+            footerIsFixed && hasScroll(scrollable) && getScrollDistanceToBottom(scrollable) > 1; // This is 1 and not 0 because a weird bug with Safari
```

Comment:
No need for checking footetIsFixed. If the value was false, then this effect returns before reaching here

Reply:
that's right